codegen/core/src/main/java/software/amazon/smithy/python/codegen/DirectedPythonCodegen.java

```diff
@@ -273,9 +274,39 @@ public void generateIntEnumShape(GenerateIntEnumDirective<GenerationContext, Pyt
 
     @Override
     public void customizeBeforeIntegrations(CustomizeDirective<GenerationContext, PythonSettings> directive) {
+        generateServiceModuleInit(directive);
+        generatePluginFiles(directive);
         generateInits(directive);
     }
 
+    /**
+     * Writes out all extra files required by runtime plugins.
+     */
+    private void generatePluginFiles(CustomizeDirective<GenerationContext, PythonSettings> directive) {
+        GenerationContext context = directive.context();
+        for (PythonIntegration integration : context.integrations()) {
+            for (RuntimeClientPlugin runtimeClientPlugin : integration.getClientPlugins(context)) {
+                if (runtimeClientPlugin.matchesService(context.model(), directive.service())) {
+                    runtimeClientPlugin.writeAdditionalFiles(context);
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates top level __init__.py file.
+     */
+    private void generateServiceModuleInit(CustomizeDirective<GenerationContext, PythonSettings> directive) {
```

Comment:
This could be done directly in `generateInits`. I'm refactoring this in a separate PR though so it's fine to have it here for now.